### PR TITLE
fix links in examples

### PR DESCRIPTION
**Pull request by Laurynas-Jagutis**
Updated the links in the example notebooks.

### CODE REVIEW

File: docs/examples/Make Test Dataset.ipynb [hunk 1]
```diff
@@ -98,7 +98,7 @@
     "### JSON Data Format\n",
     "\n",
     "The data format is well explained in these resources\n",
-    "[Serialization documentation](../user_manual/serialization.md) and some examples of Serialization are given in [Serialization notebook](./Serialization%20Example.ipynb)"
+    "[Serialization documentation](https://power-grid-model.readthedocs.io/en/stable/user_manual/serialization.html#serialization) and some examples of Serialization are given in [Serialization notebook](./Serialization%20Example.ipynb)"
```
Comment (mgovers) [inline docs/examples/Make Test Dataset.ipynb:101]: please revert this. you will need to automate adding the `https://power-grid-model.readthedocs.io/en/stable` prefix and making `md` => `html` in the workshop repo

File: docs/examples/Make Test Dataset.ipynb [hunk 2]
```diff
@@ -525,7 +525,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.11.1"
+   "version": "3.9.13"
```
Comment (mgovers) [inline docs/examples/Make Test Dataset.ipynb:528]: please don't downgrade python this much. instead, either undo this change, or make sure you have the latest python 3.11 installed and then rerun the examples using your version (preferably not in this PR though)

File: docs/examples/Power Flow Example.ipynb [hunk 3]
```diff
@@ -17,8 +17,8 @@
     " * Error handling\n",
     "\n",
     "It serves as an example of how to use the Python API. For detailed API documentation, refer to\n",
-    "[Python API reference](../api_reference/python-api-reference.md)\n",
-    "and [Native Data Interface](../advanced_documentation/native-data-interface.md).\n",
+    "[Python API reference](https://power-grid-model.readthedocs.io/en/stable/api_reference/python-api-reference.html#power-grid-model)\n",
+    "and [Native Data Interface](https://power-grid-model.readthedocs.io/en/stable/advanced_documentation/native-data-interface.html#native-data-interface).\n",
```
Comment (mgovers) [inline docs/examples/Power Flow Example.ipynb:21]: as before: the relative links were OK

File: docs/examples/Power Flow Example.ipynb [hunk 4]
```diff
@@ -65,7 +65,7 @@
     "- The attributes are of numpy array type.\n",
     "- `id` field is a unique field across the `input_data` dictionary. The keys of `input-data` are predefined.\n",
     "\n",
-    "Please refer to [Components](../user_manual/components.md) for detailed explanation of all component types and their input/output attributes."
+    "Please refer to [Components](https://power-grid-model.readthedocs.io/en/stable/user_manual/components.html#components) for detailed explanation of all component types and their input/output attributes."
```
Comment (mgovers) [inline docs/examples/Power Flow Example.ipynb:68]: as before: the relative links were OK

File: docs/examples/Power Flow Example.ipynb [hunk 5]
```diff
@@ -569,7 +569,7 @@
     "* Use the non-independent batch dataset approach whenever few parameters change per scenario, but the batch samples many different components, e.g. during N-1 tests.\n",
     "* Use the independent batch dataset approach when a dense sampling of the parameter space is desired for relatively a few different components, e.g. during time series power flow calculation\n",
     "\n",
-    "See also [performance guide](../user_manual/performance-guide.md) for the latest recommendations."
+    "See also [performance guide](https://power-grid-model.readthedocs.io/en/stable/user_manual/performance-guide.html#guidelines-for-performance-enhancement) for the latest recommendations."
```
Comment (mgovers) [inline docs/examples/Power Flow Example.ipynb:572]: as before: the relative links were OK

File: docs/examples/Serialization Example.ipynb [hunk 6]
```diff
@@ -700,7 +700,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.11.1"
+   "version": "3.9.13"
```
Comment (mgovers) [inline docs/examples/Serialization Example.ipynb:703]: as before: don't downgrade Python

File: docs/examples/Short Circuit Example.ipynb [hunk 7]
```diff
@@ -15,8 +15,8 @@
     "This notebook skips most of the common features across all calculations of power-grid-model like updating model, batch calculations and error handling.\n",
     "\n",
     "It serves as an example of how to use the Python API. For detailed API documentation, refer to\n",
-    "[Python API reference](../api_reference/python-api-reference.md)\n",
-    "and [Native Data Interface](../advanced_documentation/native-data-interface.md).\n",
+    "[Python API reference](https://power-grid-model.readthedocs.io/en/stable/api_reference/python-api-reference.html#power-grid-model)\n",
+    "and [Native Data Interface](https://power-grid-model.readthedocs.io/en/stable/advanced_documentation/native-data-interface.html#native-data-interface).\n",
```
Comment (mgovers) [inline docs/examples/Short Circuit Example.ipynb:19]: as before: the relative links were OK

File: docs/examples/Short Circuit Example.ipynb [hunk 8]
```diff
@@ -62,7 +62,7 @@
    "source": [
     "## Input Dataset\n",
     "\n",
-    "Please refer to [Components](../user_manual/components.md) for detailed explanation of all component types and their input/output attributes."
+    "Please refer to [Components](https://power-grid-model.readthedocs.io/en/stable/user_manual/components.html#components) for detailed explanation of all component types and their input/output attributes."
```
Comment (mgovers) [inline docs/examples/Short Circuit Example.ipynb:65]: as before: the relative links were OK

File: docs/examples/State Estimation Example.ipynb [hunk 9]
```diff
@@ -15,8 +15,8 @@
     "\n",
     "\n",
     "It serves an example of how to use the Python API. For detailed API documentation, refer to \n",
-    "[Python API reference](../api_reference/python-api-reference.md)\n",
-    "and [Native Data Interface](../advanced_documentation/native-data-interface.md).\n",
+    "[Python API reference](https://power-grid-model.readthedocs.io/en/stable/api_reference/python-api-reference.html#power-grid-model)\n",
+    "and [Native Data Interface](https://power-grid-model.readthedocs.io/en/stable/advanced_documentation/native-data-interface.html#native-data-interface).\n",
```
Comment (mgovers) [inline docs/examples/State Estimation Example.ipynb:19]: as before: the relative links were OK

File: docs/examples/State Estimation Example.ipynb [hunk 10]
```diff
@@ -62,7 +62,7 @@
     "Note the units of all input are standard SI unit without any prefix,\n",
     "i.e. the unit of voltage is volt (V), not kV.\n",
     "\n",
-    "Please refer [Components](../user_manual/components.md) for detailed explanation of all component types and their input/output attributes. "
+    "Please refer [Components](https://power-grid-model.readthedocs.io/en/stable/user_manual/components.html#components) for detailed explanation of all component types and their input/output attributes. "
```
Comment (mgovers) [inline docs/examples/State Estimation Example.ipynb:65]: as before: the relative links were OK

File: docs/examples/State Estimation Example.ipynb [hunk 11]
```diff
@@ -705,7 +705,7 @@
    "source": [
     "## Batch calculation\n",
     "\n",
-    "For state estimation a batch calculation can be performed in a similar manner as for powerflow calculations. Below is a short example provided for reference. For more a more detailed example about performing a batch calculation, please refer to the [Power Flow Example](Power%20Flow%20Example.ipynb#Batch-Calculation) or visit the [Documentation](../user_manual/calculations.md#batch-calculations)."
+    "For state estimation a batch calculation can be performed in a similar manner as for powerflow calculations. Below is a short example provided for reference. For more a more detailed example about performing a batch calculation, please refer to the [Power Flow Example](Power%20Flow%20Example.ipynb) or visit the [Documentation](https://power-grid-model.readthedocs.io/en/stable/user_manual/calculations.html#batch-calculations)."
```
Comment (mgovers) [inline docs/examples/State Estimation Example.ipynb:708]: as before: the relative links were OK

**Review by mgovers — CHANGES_REQUESTED** [review]
this PR should be very small and only change the sections references. all the rest should be automated in the workshop repo and readthedocs (the later already should be correct)

**Comment by TonyXiang8787** [comment]
@mgovers why do we need this? The relative links work just fine.

**Comment by mgovers** [comment]
> @mgovers why do we need this? The relative links work just fine.

yes i agree. see also my list of comments.

@Laurynas-Jagutis found a small bug where 3 links referred specifically to the `#utils` subsection of the python documentation, and we agreed to fix only those 3 links, but then this PR became much bigger.

I believe the reason is to fix the links in the workshop repo (those should point to the PGM docs, not to relative paths). But those link edits should live in the automation in the workshop repo - not here.

**Review by mgovers — APPROVED** [review]

**Pull request merged**